### PR TITLE
improvement: batch drawing of instanced meshes across sectors to reduce draw calls

### PR DESCRIPTION
**Pull request by christjt**

### CODE REVIEW

File: viewer/src/datamodels/cad/CadNode.ts [hunk 1]
```diff
@@ -45,14 +46,21 @@ export class CadNode extends THREE.Object3D {
...
   };
+  private _instancedMeshManager: InstancedMeshManager;
 
   constructor(model: CadModelMetadata, materialManager: MaterialManager) {
     super();
     this.type = 'CadNode';
     this.name = 'Sector model';
     this._materialManager = materialManager;
 
+    const instancedMeshGroup = new THREE.Group();
```
Comment (larsmoa) [inline viewer/src/datamodels/cad/CadNode.ts:57]:
```suggestion
    const instancedMeshGroup = new THREE.Group();
    instancedMeshGroup.name = 'InstancedMeshes';
```
For easier debugging

File: viewer/src/datamodels/cad/InstancedMeshManager.ts [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+ * Copyright 2021 Cognite AS
+ */
+
+import * as THREE from 'three';
+
+import { DynamicDefragmentedBuffer } from '../../utilities/datastructures/DynamicDefragmentedBuffer';
+import { MaterialManager } from './MaterialManager';
+import { InstancedMesh, InstancedMeshFile } from './rendering/types';
+
+export class InstancedMeshManager {
```
Comment (larsmoa) [inline viewer/src/datamodels/cad/InstancedMeshManager.ts:11]: Should be relatively easy, and useful, to add a unit test of this class.

**Review by larsmoa — APPROVED** [review]

File: viewer/src/datamodels/cad/InstancedMeshManager.test.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+import * as THREE from 'three';
+import { Matrix4 } from 'three';
```
Comment (larsmoa) [inline viewer/src/datamodels/cad/InstancedMeshManager.test.ts:6]: Very minor but I think we should use `THREE.Matrix4`

Comment (christjt) [reply viewer/src/datamodels/cad/InstancedMeshManager.test.ts:6]: yup, import mistake. Ill fix

**Pull request merged**